starlette/concurrency.py

```diff
@@ -1,43 +1,41 @@
-import asyncio
 import functools
-import sys
 import typing
 from typing import Any, AsyncGenerator, Iterator
 
+import anyio
+
 try:
     import contextvars  # Python 3.7+ only or via contextvars backport.
 except ImportError:  # pragma: no cover
     contextvars = None  # type: ignore
 
-if sys.version_info >= (3, 7):  # pragma: no cover
-    from asyncio import create_task
-else:  # pragma: no cover
-    from asyncio import ensure_future as create_task
 
 T = typing.TypeVar("T")
 
 
 async def run_until_first_complete(*args: typing.Tuple[typing.Callable, dict]) -> None:
-    tasks = [create_task(handler(**kwargs)) for handler, kwargs in args]
-    (done, pending) = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
-    [task.cancel() for task in pending]
-    [task.result() for task in done]
+    async def run(handler: typing.Callable[[], typing.Coroutine]) -> None:
+        await handler()
+        tg.cancel_scope.cancel()
+
+    async with anyio.create_task_group() as tg:
+        for handler, kwargs in args:
+            tg.start_soon(run, functools.partial(handler, **kwargs))
```

Comment:
The scope of `tg` is a bit confusing here, can we do it like it's done in `StreamingResponse` (and maybe use this function in `StreamingResponse`)?:
```suggestion
    async with anyio.create_task_group() as task_group:
        async def run(func: typing.Callable[[], typing.Coroutine]) -> None:
            await func()
            task_group.cancel_scope.cancel()
        
        for func, kwargs in args:
            task_group.start_soon(run, functools.partial(func, **kwargs))
```

Reply:
This diff is confusing, but I hope you get what I mean 🙈

Reply:
Sure, I can address that.  IMO I'd rather not use `run_until_first_complete` in `StreamingResponse` if the plan is to deprecate/remove it in the future.

Reply:
@JayH5  addressed in d0ca3f2bca14aa58d813e52b691b8ffd462a602b

Reply:
Cool, we'll see if/how we end up deprecating it but I'm ok with also not using it.